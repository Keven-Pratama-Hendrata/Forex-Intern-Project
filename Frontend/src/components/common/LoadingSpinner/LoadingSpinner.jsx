import React from 'react';

const LoadingSpinner = ({ 
  variant = 'default', 
  size = 'lg',
  className = '' 
}) => {
  const variants = {
    default: (
      <div className="flex min-h-screen items-center justify-center bg-gradient-to-b from-[#6ca0c1] to-[#507fa1]">
        <span className={`loading loading-spinner loading-${size}`}></span>
      </div>
    ),
    inline: (
      <span className={`loading loading-spinner loading-${size}`} />
    ),
    centered: (
      <div className="flex items-center justify-center h-full">
        <span className={`loading loading-spinner loading-${size}`}></span>
      </div>
    ),
    custom: (
      <span className={`loading loading-spinner loading-${size} ${className}`} />
    )
  };

  return variants[variant] || variants.default;
};

export default LoadingSpinner; 