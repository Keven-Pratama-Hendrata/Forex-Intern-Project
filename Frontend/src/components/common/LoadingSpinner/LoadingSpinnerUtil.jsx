/**
 * Renders the default fullscreen spinner.
 * @param {string} size Spinner size.
 * @returns {JSX.Element} Spinner element.
 */
export function renderDefaultSpinner(size) {
    return (
        <div className="flex min-h-screen items-center justify-center bg-gradient-to-b from-[#6ca0c1] to-[#507fa1]">
            <span role="status" className={`loading loading-spinner loading-${size}`}></span>
        </div>
    );
}

/**
 * Renders an inline spinner.
 * @param {string} size Spinner size.
 * @returns {JSX.Element} Spinner element.
 */
export function renderInlineSpinner(size) {
    return <span role="status" className={`loading loading-spinner loading-${size}`} />;
}

/**
 * Renders a centered spinner.
 * @param {string} size Spinner size.
 * @returns {JSX.Element} Spinner element.
 */
export function renderCenteredSpinner(size) {
    return (
        <div className="flex items-center justify-center h-full">
            <span role="status" className={`loading loading-spinner loading-${size}`}></span>
        </div>
    );
}

/**
 * Renders a custom spinner with extra className.
 * @param {string} size Spinner size.
 * @param {string} className Additional class names.
 * @returns {JSX.Element} Spinner element.
 */
export function renderCustomSpinner(size, className) {
    return <span role="status" className={`loading loading-spinner loading-${size} ${className}`} />;
}

/**
 * Renders the spinner variant based on props.
 * @param {string} variant Spinner variant.
 * @param {string} size Spinner size.
 * @param {string} className Additional class names.
 * @returns {JSX.Element} Spinner element.
 */
export function renderSpinnerVariant(variant, size, className) {
    switch (variant) {
        case 'inline':
            return renderInlineSpinner(size);
        case 'centered':
            return renderCenteredSpinner(size);
        case 'custom':
            return renderCustomSpinner(size, className);
        case 'default':
        default:
            return renderDefaultSpinner(size);
    }
} 