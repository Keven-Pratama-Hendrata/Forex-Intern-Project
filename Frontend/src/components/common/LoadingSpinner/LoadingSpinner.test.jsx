import { render } from '@testing-library/react';
import { describe, it, expect } from '@jest/globals';
import LoadingSpinner from './LoadingSpinner';

describe('LoadingSpinner Component', () => {
  it('renders correctly with default props', () => {
    const { container } = render(<LoadingSpinner />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with default variant', () => {
    const { container } = render(<LoadingSpinner variant="default" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with inline variant', () => {
    const { container } = render(<LoadingSpinner variant="inline" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with centered variant', () => {
    const { container } = render(<LoadingSpinner variant="centered" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with custom variant', () => {
    const { container } = render(
      <LoadingSpinner variant="custom" className="custom-spinner" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with different sizes', () => {
    const { container } = render(<LoadingSpinner size="xs" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with medium size', () => {
    const { container } = render(<LoadingSpinner size="md" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with large size', () => {
    const { container } = render(<LoadingSpinner size="lg" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with custom variant and size', () => {
    const { container } = render(
      <LoadingSpinner variant="custom" size="sm" className="my-custom-class" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with inline variant and custom size', () => {
    const { container } = render(
      <LoadingSpinner variant="inline" size="xl" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with centered variant and medium size', () => {
    const { container } = render(
      <LoadingSpinner variant="centered" size="md" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with invalid variant (should fallback to default)', () => {
    const { container } = render(<LoadingSpinner variant="invalid" />);
    expect(container.firstChild).toMatchSnapshot();
  });
}); 