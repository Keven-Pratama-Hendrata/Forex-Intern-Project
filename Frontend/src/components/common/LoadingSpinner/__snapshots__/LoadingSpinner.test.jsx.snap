// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`LoadingSpinner Component renders correctly with centered variant 1`] = `
<div
  class="flex items-center justify-center h-full"
>
  <span
    class="loading loading-spinner loading-lg"
  />
</div>
`;

exports[`LoadingSpinner Component renders correctly with centered variant and medium size 1`] = `
<div
  class="flex items-center justify-center h-full"
>
  <span
    class="loading loading-spinner loading-md"
  />
</div>
`;

exports[`LoadingSpinner Component renders correctly with custom variant 1`] = `
<span
  class="loading loading-spinner loading-lg custom-spinner"
/>
`;

exports[`LoadingSpinner Component renders correctly with custom variant and size 1`] = `
<span
  class="loading loading-spinner loading-sm my-custom-class"
/>
`;

exports[`LoadingSpinner Component renders correctly with default props 1`] = `
<div
  class="flex min-h-screen items-center justify-center bg-gradient-to-b from-[#6ca0c1] to-[#507fa1]"
>
  <span
    class="loading loading-spinner loading-lg"
  />
</div>
`;

exports[`LoadingSpinner Component renders correctly with default variant 1`] = `
<div
  class="flex min-h-screen items-center justify-center bg-gradient-to-b from-[#6ca0c1] to-[#507fa1]"
>
  <span
    class="loading loading-spinner loading-lg"
  />
</div>
`;

exports[`LoadingSpinner Component renders correctly with different sizes 1`] = `
<div
  class="flex min-h-screen items-center justify-center bg-gradient-to-b from-[#6ca0c1] to-[#507fa1]"
>
  <span
    class="loading loading-spinner loading-xs"
  />
</div>
`;

exports[`LoadingSpinner Component renders correctly with inline variant 1`] = `
<span
  class="loading loading-spinner loading-lg"
/>
`;

exports[`LoadingSpinner Component renders correctly with inline variant and custom size 1`] = `
<span
  class="loading loading-spinner loading-xl"
/>
`;

exports[`LoadingSpinner Component renders correctly with invalid variant (should fallback to default) 1`] = `
<div
  class="flex min-h-screen items-center justify-center bg-gradient-to-b from-[#6ca0c1] to-[#507fa1]"
>
  <span
    class="loading loading-spinner loading-lg"
  />
</div>
`;

exports[`LoadingSpinner Component renders correctly with large size 1`] = `
<div
  class="flex min-h-screen items-center justify-center bg-gradient-to-b from-[#6ca0c1] to-[#507fa1]"
>
  <span
    class="loading loading-spinner loading-lg"
  />
</div>
`;

exports[`LoadingSpinner Component renders correctly with medium size 1`] = `
<div
  class="flex min-h-screen items-center justify-center bg-gradient-to-b from-[#6ca0c1] to-[#507fa1]"
>
  <span
    class="loading loading-spinner loading-md"
  />
</div>
`;
